README.md

```diff
@@ -34,18 +28,53 @@ npm install react-image --save
 
 ## Dependencies
 
-`react-image` has no external dependencies, aside for the usual `react` and `react-dom`.
+`react-image` has no external dependencies, aside for a version of `react` and `react-dom` which support hooks.
 
 ## Documentation
 
+You can use the standalone component, documented below, or the `useImage` hook.
+
+### useImage():
+
+The `useImage` hook allows for incorperating `react-image`'s logic in any component. When using the hook, the component can be wrapped in `<Suspense>` to keep it from rendering until the image is ready. Specify the `fallback` prop to show a spinner or any other component to the user while the browser is loading. The hook the throw an error if it failes to find any images. You can wrap your componenet with an [Error Boundry](https://reactjs.org/docs/code-splitting.html#error-boundaries) to catch this scenario and do/show something.
+
+Example usage:
+
+```js
+import useImge from 'react-image'
+
+export default function MyComponent() {
+  const {src, isLoading, error} = useImage({
+    srcList: 'https://www.example.com/foo.jpg',
```

Comment:
another thing with this example is that it wont work unless i set `useSuspense: false`

it will throw an error.

either update to show a useSuspense example or show setting it to false

Reply:
What is the error?

Reply:
```
react-dom.development.js:20938 Uncaught Error: App suspended while rendering, but no fallback UI was specified.

Add a <Suspense fallback=...> component higher in the tree to provide a loading indicator or placeholder to display.
    in App
    at throwException (react-dom.development.js:20938)
    at handleError (react-dom.development.js:22010)
    at performSyncWorkOnRoot (react-dom.development.js:21759)
    at scheduleUpdateOnFiber (react-dom.development.js:21188)
    at updateContainer (react-dom.development.js:24373)
    at react-dom.development.js:24758
    at unbatchedUpdates (react-dom.development.js:21903)
    at legacyRenderSubtreeIntoContainer (react-dom.development.js:24757)
    at Object.render (react-dom.development.js:24840)
    at Module../src/app.js (app.js:162)
```

Reply:
Can you please show the code you used?

Reply:
i juat added to app.js

```js

import useImage from '../src/useImage.js'

...

const {src, isLoading, error} = useImage({
    srcList: [
      'https://www.exampleblacjajkdld.com/foo.jpg',
      'https://www.exampleblacjajkdld.com/foo.jpg',
      'https://www.exampleblacjajkdld.com/foo.jpg',
      'https://www.exampleblacjajkdld.com/foo.jpg',
      'https://www.example.com/foo.jpg',
      'https://via.placeholder.com/150', // will be loaded
    ],
   // useSuspense: false,
  })

...

return (
...

<div>
        <h5>using hooks</h5>
        <img src={src} />
      </div>
);
```

Reply:
Could it be the component wasn't wrapped in Suspense? I've added examples for hooks, you can run `npm i && npm run dev` to see

Reply:
yea its clear now